includes/tracking-events.php

```diff
@@ -0,0 +1,120 @@
+            $this->put_event_in_queue(
+                "window.metrilo.search('"
+                . get_search_query()
+                . "', '"
+                . get_search_link()
+                . "');"
+        
+            );
+            $this->single_item_tracked = true;
+        }
+        
+        // if visitor is viewing shopping cart page
+        if(!$this->single_item_tracked && is_cart()){
+            $this->put_event_in_queue(
+                "window.metrilo.customEvent('view_cart');"
+            );
+            $this->single_item_tracked = true;
+        }
+        // if visitor is anywhere in the checkout process
+        if (!$this->single_item_tracked && is_order_received_page()) {
+            $this->put_event_in_queue(
+                "window.metrilo.viewPage('"
+                . get_permalink()
+                . "', "
+                . json_encode(['name' => get_the_title()])
+                . ");"
+            );
+            $this->single_item_tracked = true;
+            
+        } elseif (
+            !$this->single_item_tracked &&
+            function_exists('is_checkout_pay_page') &&
+            is_checkout_pay_page()
+        ) {
+            $this->put_event_in_queue(
+                "window.metrilo.customEvent('checkout_payment');"
+            );
+            $this->single_item_tracked = true;
+        } elseif (!$this->single_item_tracked && is_checkout()) {
+            $this->put_event_in_queue(
+                "window.metrilo.checkout();"
+            );
+            $this->single_item_tracked = true;
+        }
+    }
+    //
+    //  ** GENERIC WordPress tracking - doesn't require WooCommerce in order to work **//
+    //
+    //  if visitor is viewing homepage or any text page
+    if (!$this->single_item_tracked && is_front_page()) {
+        $this->put_event_in_queue(
+            "window.metrilo.viewPage('"
+            . get_permalink()
+            . "', "
+            . json_encode(['name' => get_the_title()])
+            . ");"
+        );
+        $this->single_item_tracked = true;
+    } elseif (!$this->single_item_tracked && is_page()) {
+        $this->put_event_in_queue(
+            "window.metrilo.viewPage('"
+            . get_permalink()
+            . "', "
+            . json_encode(['name' => get_the_title()])
+            . ");"
+        );
+        $this->single_item_tracked = true;
+    }
+
+    //  if visitor is viewing post
+    if (!$this->single_item_tracked && is_single()) {
+        $post_id = get_the_id();
+        $this->put_event_in_queue(
+            "window.metrilo.viewArticle('"
+            . $post_id
+            . "', "
+            . json_encode(['name' => get_the_title(), 'url' => get_permalink($post_id)])
+            . ");"
+        );
+        $this->single_item_tracked = true;
+    }
+    
+    //  check if there are events in the queue to be sent to Metrilo
+    if ($this->identify_call_data !== false) {
```

Comment:
Can it be `null`?

Reply:
Could this one help here? https://www.php.net/manual/en/function.empty.php

Reply:
This is something i forgot to do.. the identify call. Will add it as new review after we are done here.

Reply:
Please, do share a link in this discussion with the changes that will be done, regarding this piece of code.

Reply:
No need for check against boolean. The identify_call_data is either false or email string. Will change it to if($this->identify_call_data)

Reply:
> This is something i forgot to do.. the identify call. Will add it as new review after we are done here.

Is this going to happen after this pull request is merged?

Reply:
This is something that already happened in the previous commits.